### PR TITLE
Vote article

### PR DESCRIPTION
**Pull request by SamirabintABS**
increment and decrement votes
Error handling

### CODE REVIEW

File: src/components/VoteButton.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        // })
+        // api.voteArticle(id)
+        //     .then((updatedArticle) => {
+        //     }).catch(() => {
+        //         setArticle((currentArticle) => {
+        //             return { ...currentArticle, votes: currentArticle.votes - 1 }
+        //         })
+        //         setErr("something went wrong, try again later")
+        //     })
+    }
```
Comment (popatre) [inline src/components/VoteButton.jsx]: Could you get rid of any comments. 
Comments are one of the worst to leave in a project as no one else will delete them as they presume someone else needs them, or they're there for a reason

File: src/components/VoteButton.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const id = articleId.article_id;
+    const [err, setErr] = useState("")
+
+    const handleClick = () => {
+        if (addedVotes === 0) {
+            setAddedVotes(1);
+            api.voteArticle(id, 1).catch(() => {
+                setErr("something went wrong, try again later")
+                setAddedVotes(0)
+            })
```
Comment (popatre) [inline src/components/VoteButton.jsx:15]: Nice optimistic rendering of the votes

File: src/components/VoteButton.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        }
+
+    }
+
+
+    return (
+        <div>
+            <button onClick={handleClick} aria-label="Vote for this article" className="secondary-btn">Vote!</button>
+            {err ? <p>{err}</p> : null}
```
Comment (popatre) [inline src/components/VoteButton.jsx]: Could neaten this to `err && <p>{err} </p>`

**Review by popatre — APPROVED** [review]
Looks great, well done!

**Comment by NRMSMITH** [comment]
Looks fab - merge ! 😄 

Also I'm obsessed with the frog

**Pull request merged**